compaction/compaction.cc

```diff
@@ -772,8 +772,8 @@ class compaction {
     // compact_mutation_v2. it just uses a single compaction writer for writing
     // the compacted fragments.
     future<> compact_on_writer_without_gc() {
-        auto consumer = make_interposer_consumer([this] (flat_mutation_reader_v2 reader) mutable {
```

Comment:
probably append _and_interposer_consumer to function name so to make it explicit as to which condition we want this procedure to be used.

Reply:
@raphaelsc hi Raphael, after a second thought, i think, to specialize `compact_on_writer_without_gc()` as separate function does not help with the readability. it is not *designed* to handle the case where we don't use interpose consumer and we are not allowed to use gc consumer. it *happens* to be put on this spot,  just because the reader side compaction has a higher priority than it. so i restructured code so it looks more symmetric.

but i still don't get why the reader side compaction relies on / requires `use_interposer_consumer()` . could help me understand why they are connected? IMHO, to use interposer consumer or not is only used as the indication when the base compaction chooses to use the end consumer or the interposed one provided by the derived class.